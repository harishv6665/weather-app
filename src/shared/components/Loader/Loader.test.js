import React from "react";
import { shallow } from "enzyme";
import {Loader} from "../";

describe("Loader tests", () => {

    it("should be defined", () => {
        expect(Loader).toBeDefined();
    });

    it("should match to snapshot", () => {
        const wrapper = shallow(<Loader />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should match the DOM structure", () => {
        const wrapper = shallow(<Loader />);
        expect(wrapper.find("div").length).toBe(2);
        expect(wrapper.find("div div").length).toBe(1);
        expect(wrapper.find("span").length).toBe(1);
        expect(wrapper.find("div span").length).toBe(1);
        expect(wrapper.find("span").text()).toEqual("Loading...");
    });

});
