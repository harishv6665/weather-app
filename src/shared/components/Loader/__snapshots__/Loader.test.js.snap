// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader tests should match to snapshot 1`] = `ShallowWrapper {}`;
